presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -144,17 +151,41 @@ public boolean isForceLocalScheduling()
         return columnCoercions;
     }
 
+    public InternalHiveBlock currentBlock()
+    {
+        checkState(!isDone(), "All blocks have been consumed");
+        return blocks.get(currentBlockIndex);
+    }
+
+    public boolean isDone()
+    {
+        return currentBlockIndex == blocks.size();
+    }
+
+    public void increaseStart(long value)
+    {
+        start += value;
+        if (start == currentBlock().getEnd()) {
+            currentBlockIndex++;
+            if (isDone()) {
+                return;
+            }
+            checkArgument(start == currentBlock().getStart());
+        }
+    }
+
     public int getEstimatedSizeInBytes()
     {
+        // TODO! revisit
```

Comment:
Add more detail or remove this comment. It's not useful to readers in the current form